content/blog/sivabala-opsql-opsramps-kalpavriksha-wish-fulfilling-tree.md

```diff
@@ -0,0 +1,194 @@
+
+
+
+## Basic Syntax and Structure
+
+The general syntax for an OpsQL query is straightforward:
+
+```PlainText
+ <attribute> <operator> | <coperator> "<value>" [[<operator> [<attribute> | <coperator> "<value>"[)]] ... ]
+```
+
+You can use logical operators  `AND` and `OR` to refine your search further.
+
+## Operators
+OpsQL supports a variety of operators to create precise queries.
+### Equality Operators
+ `=`, `!=` 
+### String Operators
+ `CONTAINS`, `STARTS WITH`, `LIKE`
+### Logical Operators
+`AND`, `OR`
+
+For example, to find all resources with an agent installed and of type "Windows," you would use:
+
+```PlainText
+agentInstalled = "true" AND type = "Windows"
+```
+
+## Attributes
+Attributes are different types of information available on an object. For instance, a resource might have attributes like `make`, `ipAddress`, and `agentInstalled`, while an alert might have attributes like `priority`, `currentState`, and `createdTime`.
+
+
+
+For more details, you can refer to the [OpsRamp Documentation](https://docs.opsramp.com/platform-features/feature-guides/query-language-reference/query-language-ref/)(https://docs.opsramp.com/platform-features/feature-guides/query-language-reference/query-language-ref/ "Query Language Reference | OpsRamp Documentation").
+
+## More Examples
+### Search for resources that were discovered by an AWS Integration​
+
+```PlainText
+installedAppName = aws
+```
+
+### Search for resources that I tagged in AWS with AWS tag OwnerName :SivaBalaSubramanian
+```PlainText
+installedAppName = "aws" AND tags.name = "OwnerName" and tags.value = "SivaBalaSubramanian"​
+```
+### Search for resources with alerts that have been open for the last 2 hours​
+
+```PlainText
+createdTime > "-7200sec" ​
+createdTime > "-120min"​
+```
+
+### Search for Open alerts that are Critical​
+```PlainText
+currentState = "CRITICAL" AND status = "OPEN" ​
+```
+​
+### Search for Open and Critical alerts that have an Open incident​
+```PlainText
+currentState = "CRITICAL" AND status = "OPEN" AND incidentId IS NOT NULL ​
+```
+
+### Search for alerts that have been Open for longer than 2 hours​
+```PlainText
+createdTime < "-7200sec" ​
+``` 
+​
+### Search for Open and Critical alerts on resources tagged with AWS tag “BU: bu-123”​
+```PlainText
+currentState = "CRITICAL" AND status = "OPEN" AND resource.tags.name = "Team" AND resource.tags.value = "Opsqa"​
+```
+
+## What is OpsQL API?
+The OpsQL API is a powerful interface that allows users to execute OpsQL queries programmatically. This API provides the flexibility to filter, search data within the OpsRamp platform, making it an indispensable tool for IT administrators and developers.
+

```

Comment:
You may want to add a brief description of the tools that readers/developers/IT Administrators can use to make OpsQL API calls such as Postman, cURL, Python for example.
I believe in the next section you provide readers with Python example to make OpsQL API calls.